### ISSUE TITLE
Problems with conch examples

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4358]
|<img alt="cratuki's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| cratuki reported|
|-|-|
|Trac ID|trac#4358|
|Type|defect|
|Created|2010-03-08 03:19:28Z|

I was looking at conch examples found at http://twistedmatrix.com/documents/current/conch/examples/#auto2 on 20100307.

Problems:

1) Numbered links in first section of page don't navigate to new pages.

2) demo_draw doesn't work. It mentions insult dependencies in the stack trace. This may have been modified when someone thought they were working on demo_insults.trac.

3) For the last four, the service starts, but a telnet to the port gives the following output:
```
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
SSH-2.0-Twisted
####
    #f#ju##p#
             1diffie-hellman-group1-sha1ssh-rsa#aes256-ctr,aes256-cbc,aes192-ctr,aes192-cbc,aes128-ctr,aes128-cbc,cast128-ctr,cast128-cbc,blowfish-ctr,blowfish-cbc,3des-ctr,3des-cbc#aes256-ctr,aes256-cbc,aes192-ctr,aes192-cbc,aes128-ctr,aes128-cbc,cast128-ctr,cast128-cbc,blowfish-ctr,blowfish-cbc,3des-ctr,3des-cbchmac-sha1,hmac-md5hmac-sha1,hmac-md5     none,zlib       none,zlib#;Hk
```

It's unclear how to proceed past this.

4) It would be good to mention dependencies for the scripts in the files. I did this to get them working:
. easy_install pyCrypto
. easy_install pyasn1

Attachments:

* [irm.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/812/812bb5f95eea838a452d940c641316c530676172/c3780eb57f80d02b1ae7bbf7b4fe13438be959c3.patch) (499 bytes) - added by exarkun on 2010-03-14 15:27:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4358 4358
type__defect defect
reporter__cratuki cratuki
priority__normal normal
milestone__ 
branch__ 
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1268018368000000 1268018368000000
changetime__1269213081000000 1269213081000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to z3p|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

  1. They're links to anchors on the same page.
  1. r14200 removed the module level `IRM` constant, breaking `demo_draw.tac`.
  1. That's the SSH port.
  1. `twisted/conch/topfiles/README` enumerates Conch's dependencies.  I don't think we should duplicate the dependency information in N example files.  Other suggestions welcome, though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Attached fixes the example.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @exarkun|
|-|-|

I don't have the Crypto dependency to run the example right now, but assuming that it now runs without any problems, the change looks good. demo_draw.tac needs a copyright bump and moving the triple quotes to lines of their own, but other than that looks good to apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [28765]) Apply irm.patch fixing the insults drawing example

Author: exarkun
Reviewer: jesstess
Fixes: [#4358](https://github.com/twisted/twisted/issues/4358)

Fix the usage of an insults API in the drawing example which was broken quite a long time ago.
The new code uses the correct name for a constant to enable insertion replacement mode.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
